tools/oeedger8r/Emitter.ml

```diff
@@ -854,24 +854,29 @@ let warn_non_portable_types (fd:Ast.func_decl) =
   (if uses_type (Ast.Long Ast.Unsigned) fd || uses_type ulong_t fd then
      print_portability_warning_with_recommendation "unsigned long" "uint64_t or uint32_t")
 
-let warn_signed_size_types (fd:Ast.func_decl) =
+let warn_signed_size_or_count_types (fd:Ast.func_decl) =
   let print_signedness_warning ty =
-    printf "Warning: Function '%s': Size parameter '%s' should not be signed.\n" fd.fname ty
+    printf "Warning: Function '%s': Size or count parameter '%s' should not be signed.\n" fd.fname ty
   in
-  (* Get the names of all size parameters for the function [fd]. *)
+  (* Get the names of all size and count parameters for the function [fd]. *)
   let size_params = List.map (fun (ptype, decl) ->
+      (* The size may be either a [count] or [size], and then either a
+         number or string. We are interested in the strings, as the
+         indicate named [size] or [count] parameters. *)
+      let param_name { ps_size; ps_count } =
+        match ps_size, ps_count with
+        (* [s] is the name of the parameter as a string. *)
+        | (None, Some (Ast.AString s) | Some (Ast.AString s), None) -> s
+        (* TODO: Check for [Some (Ast.ANumber n)] that [n > 0] *)
+        | _ -> ""
+      in
+      (* Only variables that are pointers where [chkptr] is true may
+         have size parameters. TODO: Validate this! *)
       match ptype with
-      (* Only variables that are pointers where [chkptr] is true may have
-         size parameters. TODO: Validate this! *)
-      | Ast.PTPtr (atype, ptr_attr) when ptr_attr.Ast.pa_chkptr->
-        (* The size may be either a count, [ANumber], or named size
-           parameter, [AString]. *)
-        (match ptr_attr.Ast.pa_size.Ast.ps_size with
-         | None -> ""
-         | Some (Ast.ANumber n) -> "" (* TODO: Check that [n > 0] *)
-         | Some (Ast.AString s) -> s (* We have the name of the parameter. *))
+      | Ast.PTPtr (atype, ptr_attr) when ptr_attr.Ast.pa_chkptr ->
```

Comment:
@anakrish This part I have the least understanding of since it's an implementation detail of EDL. Can you check this for me? I pulled it out of `oe_prepare_input_buffer` where we were calling `oe_get_param_size`.

Reply:
@Andy, Looks good to me. We are finding out the names of parameters used as size/count for other parameters.

Note: EDL allows both size *and* count to be specified for a parameter; we don't support it yet.

Reply:
No, no, I got that (although that note is useful, what a strange feature; I'll add a warning for now). I am not sure about `when ptr_attr.Ast.pa_chkptr`. I want to validate that my understanding is correct: only pointers where `pa_chkptr` is true (and all of those pointers) are pointers that will need a size and/or count specified.

Reply:
(The only comment around it is "Whether to generate code to check pointer.")

Reply:
Yes, if [user_check] is specified, then the edger8r does not do anything with a given pointer parameter.
In that case the pa_chkptr will be false. 
Your logic looks good to me.